src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -32,17 +36,26 @@ public override TextDocumentAttributes GetTextDocumentAttributes(DocumentUri uri
 
         public override Task<Unit> Handle(DidChangeTextDocumentParams request, CancellationToken token)
         {
+            DocumentUri documentUri = request.TextDocument.Uri;
             // we have full sync enabled, so apparently first change is the whole document
             var contents = request.ContentChanges.First().Text;
 
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, contents);
+            if (string.Equals(Path.GetFileName(documentUri.Path), LanguageConstants.BicepConfigSettingsFileName))
+            {
+                // Refresh compilation of source files in workspace when local bicepconfig.json file is edited
```

Comment:
IMO - for consistency with modules, we should. We can always look into https://github.com/Azure/bicep/issues/3787 for perf improvements, though it feels like less of a concern than reloading the config file on every keypress in a `.bicep` file.

Reply:
I'm not so sure. `.bicep` files and `bicepconfig.json` serve different purposes and don't have to be consistent. I'm not sure there's precent for recompiling the world on config file changes in other frameworks, either.

It's not only a perf concern. As you're editing the config file, it will bounce between valid and invalid state. This means that the diagnostics we emit for all the modules that are open (and possibly all the the modules in the workspace in the future) will be bouncing between "whatever gets emitted by default" and "whatever gets emitted with the specified config".

Reply:
If we did it on save, you'd just get one update to diagnostics that's consistent with what you changed.

Reply:
> It's not only a perf concern. As you're editing the config file, it will bounce between valid and invalid state. This means that the diagnostics we emit for all the modules that are open (and possibly all the the modules in the workspace in the future) will be bouncing between "whatever gets emitted by default" and "whatever gets emitted with the specified config".

That's a really good point, I'm convinced!

Reply:
Reverted changes here:
https://github.com/Azure/bicep/commit/1ee10711ad7bdd7412967c1ce6ae6f6a1923fa5b

We will only be refreshing compilation when bicepconfig.json file is created, deleted or saved. Let me know if you have any questions or concerns.

Reply:
Sounds good to me.